File: online_judge/LC/1518/1518.cpp
class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        int num_empty_bottle = numBottles;
        int total_drink = numBottles;
        while (num_empty_bottle >= numExchange) {
            int num_change = num_empty_bottle / numExchange;
            total_drink += num_change;
            num_empty_bottle = num_empty_bottle - num_change * numExchange + num_change;
        }

        return total_drink;
    }
};

File: LC/1518/1518.py
# link of the question:https://leetcode.com/problems/water-bottles/

class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        num_empty_bottle = numBottles
        max_bottle = numBottles
        while True:
            if num_empty_bottle < numExchange:
                return max_bottle
            else:
                num_change = num_empty_bottle // numExchange
                num_empty_bottle = num_empty_bottle - num_change * numExchange + num_change
                max_bottle += num_change